tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/SdkRepoConfigHelper.cs

```diff
@@ -0,0 +1,210 @@
+                }
+
+                _logger.LogDebug("Retrieved config value from {JsonPath} as {Type}", jsonPath, typeof(T).Name);
+                return value;
+            }
+            catch (JsonException ex)
+            {
+                throw new InvalidOperationException($"Error parsing JSON configuration: {ex.Message}", ex);
+            }
+        }
+
+        // Get build configuration (either command or script path)
+        public async Task<(BuildConfigType type, string value)> GetBuildConfigurationAsync(string repositoryRoot)
+        {
+            // Try command first
+            try
+            {
+                var command = await GetConfigValueFromRepoAsync<string>(repositoryRoot, BuildCommandJsonPath);
+                if (!string.IsNullOrEmpty(command))
+                {
+                    _logger.LogDebug("Found build command configuration");
+                    return (BuildConfigType.Command, command);
+                }
+            }
+            catch (InvalidOperationException)
+            {
+                // Command not found, continue to try path
+                _logger.LogDebug("No build command found, trying script path");
+            }
+
+            // Try path
+            try
+            {
+                var path = await GetConfigValueFromRepoAsync<string>(repositoryRoot, BuildScriptPathJsonPath);
+                if (!string.IsNullOrEmpty(path))
+                {
+                    _logger.LogDebug("Found build script path configuration");
+                    return (BuildConfigType.ScriptPath, path);
+                }
+            }
+            catch (InvalidOperationException)
+            {
+                // Path not found either
+                _logger.LogError("No build configuration found");
+            }
+
+            throw new InvalidOperationException($"Neither '{BuildCommandJsonPath}' nor '{BuildScriptPathJsonPath}' found in configuration.");
+        }
+
+        // Substitute template variables in command strings
+        public string SubstituteCommandVariables(string command, Dictionary<string, string> variables)
+        {
+            if (string.IsNullOrEmpty(command))
+            {
+                return command;
+            }
+
+            if (variables == null || variables.Count == 0)
+            {
+                return command;
+            }
+
+            var result = command;
+            
+            // Replace variables in the format {variableName}
+            foreach (var variable in variables)
+            {
+                var placeholder = $"{{{variable.Key}}}";
+                result = result.Replace(placeholder, variable.Value, StringComparison.OrdinalIgnoreCase);
+            }
+
+            _logger.LogDebug("Command after variable substitution: {Result}", result);
+            return result;
+        }
+
+        // Parse command string into command and arguments
+        public string[] ParseCommand(string command)
```

Comment:
System.CommandLine also exposes its parser if you think that would be more robust:

```
var parser = new Parser(new RootCommand());  // necessary to access .Parse()
var result = parser.Parse("cmd --flag \"quoted arg\"");
foreach (var t in result.Tokens)
{
    Console.WriteLine(t.Value);
}
```

Reply:
Sounds good and migrated code to use it.